src/correlation.rs

```diff
@@ -0,0 +1,157 @@
+use ndarray::prelude::*;
+use ndarray::Data;
+use num_traits::{Float, FromPrimitive};
+
+pub trait CorrelationExt<A, S>
+where
+    S: Data<Elem = A>,
+{
+    /// Return the covariance matrix `C` for a 2-dimensional
+    /// array of observations `M`.
+    ///
+    /// Let `(r, o)` be the shape of `M`:
+    /// - `r` is the number of random variables;
+    /// - `o` is the number of observations we have collected 
+    /// for each random variable.
+    /// 
+    /// Every column in `M` is an experiment: a single observation for each 
+    /// random variable.
+    /// Each row in `M` contains all the observations for a certain random variable.
+    /// 
+    /// The parameter `ddof` specifies the "delta degrees of freedom". For
+    /// example, to calculate the population covariance, use `ddof = 0`, or to
+    /// calculate the sample covariance (unbiased estimate), use `ddof = 1`.
+    ///
+    /// The covariance of two random variables is defined as:
+    ///
+    /// ```text
+    ///                1       n
+    /// cov(X, Y) = ――――――――   ∑ (xᵢ - x̅)(yᵢ - y̅)
+    ///             n - ddof  i=1
+    /// ```
+    ///
+    /// where
+    ///
+    /// ```text
+    ///     1   n
+    /// x̅ = ―   ∑ xᵢ
+    ///     n  i=1
+    /// ```
+    /// and similarly for ̅y. 
+    ///
+    /// **Panics** if `ddof` is greater than or equal to the number of observations.
```

Comment:
It's worth noting that the implementation also panics if converting the number of observations to `A` with `FromPrimitive` fails.